gst/datarepo/gstdatareposink.c

```diff
@@ -182,20 +227,21 @@ gst_data_repo_sink_get_property (GObject * object, guint prop_id,
 }
 
 /**
- * @brief Called when a buffer should be presented or ouput.
+ * @brief Function to write others media type (tensors(fixed), video, audio, octet and text)
  */
 static GstFlowReturn
-gst_data_repo_sink_render (GstBaseSink * bsink, GstBuffer * buf)
+gst_data_repo_sink_write_others (GstDataRepoSink * sink, GstBuffer * buffer)
 {
-  GstDataRepoSink *sink = GST_DATA_REPO_SINK_CAST (bsink);
   gsize write_size = 0;
   GstMapInfo info;
   guint to_write = 0, byte_write = 0;
 
+  g_return_val_if_fail (sink != NULL, GST_FLOW_ERROR);
+  g_return_val_if_fail (buffer != NULL, GST_FLOW_ERROR);
   g_return_val_if_fail (sink->fd != 0, GST_FLOW_ERROR);
 
   GST_OBJECT_LOCK (sink);
-  gst_buffer_map (buf, &info, GST_MAP_READ);
+  gst_buffer_map (buffer, &info, GST_MAP_READ);
```

Comment:
line 244 and 331 update the memory block, need gst_map_write?

Reply:
only need to read data from the GstMapInfo.data.
Thank you.